packages/amplify-e2e-tests/src/__tests__/storage-1.test.ts

```diff
@@ -57,6 +59,14 @@ describe('amplify add/update storage(S3)', () => {
     await validate(projRoot);
   });
 
+  it('init a project and add S3 bucket with hyphenated-name and Auth only access', async () => {
```

Comment:
I don't think we should get in the habit of writing an e2e test for input validation. A unit test should be sufficient for this

Reply:
This test is to ensure that the hyphenated bucket names created through amplify are correctly pushed. Do you feel it's better to remove this E2E to reduce the amount of time spent in the deployment pipelines?

Reply:
I agree with @edwardfoyle - I think that this is logic that can be tested by a unit test. The behavior isn't impacted by integration of other amplify commands/AWS services, so a unit test should suffice.

Reply:
This particular test was added because, there is some logic in the S3 workflow (legacy) which appends information prior to writing to Cloudformation by using hyphens as delimiters in the bucket name. Any future change to this part  of the code will not get caught with Snapshot tests . I can remove this if it feels not required.